test/functional/test-xhr.js

```diff
@@ -381,7 +381,8 @@ describe.configure().skipSafari().run('XHR', function() {
         });
       });
 
-      it('should do simple JSON fetch', () => {
+      // TODO(#16916): Make this test work with synchronous throws.
+      it.skip('should do simple JSON fetch', () => {
```

Comment:
I have no explanation why but after this, it silently fails rest of the tests in the file.

What should have been an execution of 100 tests altogether, are now just 9 tests but passing

results via locally running `gulp test --files=test/functional/test-xhr.js --watch`

Reply:
If you hit the debug button in the karma window, do you see any errors when the tests are re run?

Reply:
@prateekbh I was able to repro your problem. One of your tests is throwing an error in `beforeEach()`. Mocha considers this a catastrophic error and bails. Let me know if you need help reproing it.

<img width="1312" alt="" src="https://user-images.githubusercontent.com/26553114/43332032-1ea4c476-9195-11e8-8d41-5e8af678c918.png">

Reply:
well i can use some help on how to find whats the issue, I am creating a PR which just toucher `xhr-impl.js` and hence I reproduced this error. 
Not really sure about the setup and test cases here in `functional`